### DISCUSSION TITLE
How to work properly with luminosities computed for a given filter?

### DISCUSSION THREAD

**Discussion by liempi** [galacticusorg/galacticus#697] [Q&A] [answered]
Hi all.

I'm new working with the luminosities computed by Galacticus, and I think it could be useful to ask here to keep the answer visible.

I computed the luminosities using in a _rest_ (and _observed_) frame using the `JWST_NIRCAM_f090w` filter available.  
Then Galacticus outputs the stellar luminosities at a given redshift in units of W/Hz according to the physics manual. So,  what is the best approach to obtain the luminosity in watts? 

Many thanks :)

**Comment by abensonca** [comment, marked as answer]
A few important points here:

1. The luminosities are _not_ output in units of W/Hz. They're output in units of $4.4659 \times 10^{13}$ W/Hz which is the zero point of the [AB magnitude system](https://en.wikipedia.org/wiki/AB_magnitude). 
2. Usually, in astronomical observations, you don't want the actual luminosity (e.g. the luminosity in Watts as you are asking about), because luminosities are almost always measured through a filter (often the combination of an actual filter and the detector response etc.). That's why the usual quantity measured is the luminosity relative to a source of constant luminosity density observed through the same filter, i.e.:

$$
\tilde{L} = \left. \int_0^\infty \frac{S(\nu)}{\mathrm{h}\nu} R(\nu) \mathrm{d}\nu \right/  \int_0^\infty \frac{S_\mathrm{AB}}{\mathrm{h}\nu} R(\nu) \mathrm{d}\nu 
$$

where $S(\nu)$ is the spectral density (W/Hz) of the source, $S_\mathrm{AB}$ is the spectral density for the AB magnitude system, and $R(\nu)$ is the filter response (fraction of photons that pass through the filter. Note that the $\mathrm{h}\nu$ term is included as most (at least most modern) sensors count photons.
3. If you _really_ want a luminosity - a thing measured in W, you need to first think carefully about what you actually want. You could define some luminosity $\hat{L}$:

$$
\hat{L} = \tilde{L} \mathrm{h}\bar{\nu} \int_0^\infty \frac{S_\mathrm{AB}}{\mathrm{h}\nu} R(\nu) \mathrm{d}\nu  =\mathrm{h}\bar{\nu} \  \int_0^\infty \frac{S(\nu)}{\mathrm{h}\nu} R(\nu) \mathrm{d}\nu 
$$

where $\bar{\nu}$ is some mean frequency in the filter. This isn't very meaningful though - it depends on the filter shape (and, this is exactly _why_ we typically don't measure these).

If you really wanted to know the total luminosity in some frequency range you could define a filter that was proportional to $\nu$ in that range and zero outside, the integral would then effectively become:


$$
\tilde{L} = \left. \int_{\nu_\mathrm{min}}^{\nu_\mathrm{max}} S(\nu) \mathrm{d}\nu \right/  \int_{\nu_\mathrm{min}}^{\nu_\mathrm{max}}  S_\mathrm{AB} \mathrm{d}\nu  = \left. \int_{\nu_\mathrm{min}}^{\nu_\mathrm{max}} S(\nu) \mathrm{d}\nu \right/ (S_\mathrm{AB} [\nu_\mathrm{max} - \nu_\mathrm{min}])
$$

which you could then multiply by $S_\mathrm{AB} [\nu_\mathrm{max} - \nu_\mathrm{min}]$ to get the total luminosity. This is not a quantity you can compared to anything observed though, so I doubt that it's useful.

So, what are you trying to accomplish with the luminosities that you are outputting? If you explain that in more detail I can probably suggest the best solution.

**Reply by liempi** [reply]
I want to compute the color of a galaxy given two filters and also construct the luminosity function.

**Reply by abensonca** [reply]
So, probably what Galacticus is outputting is exactly what you want! Colors are typically defined in terms of magnitudes. If the luminosity as output by Galacticus is $L$, then the corresponding absolute magnitude is just $M = - 2.5 \log_{10} L$. The color of the object is then just the difference in the magnitude between the two filters.

For the luminosity function, is there specific data you're trying to compare to?

**Reply by liempi** [reply]
Many thanks!

For the luminosity function there is no dataset yet, but I plan to compare with the [this work](https://ui.adsabs.harvard.edu/abs/2024arXiv240610341A/abstract), where the AGN bolometric luminosity function is computed for some galaxies which I want to select from a color-magnitude criteria.

**Reply by abensonca** [reply]
I see. Selecting from a color-magnitude criterion should be straightforward in that case.

**Reply by liempi** [reply]
Yes, it was unclear to me how to start working with the luminosities, but now the procedure is clear. 

Many thanks again!
